ACadSharp/IO/DWG/DwgStreamReaders/DwgObjectReader.cs

```diff
@@ -3048,23 +3048,25 @@ private CadTemplate readMultiLeader()
 				//	- Scaling (using scale vector)
 				//	- Translation (using location)
 				//	Simply read an array of 16 doubles:
-				double[] matrix  = annotContext.TransformationMatrix;
-				matrix[0] = _objectReader.ReadBitDouble();
-				matrix[1] = _objectReader.ReadBitDouble();
-				matrix[2] = _objectReader.ReadBitDouble();
-				matrix[3] = _objectReader.ReadBitDouble();
-				matrix[4] = _objectReader.ReadBitDouble();
-				matrix[5] = _objectReader.ReadBitDouble();
-				matrix[6] = _objectReader.ReadBitDouble();
-				matrix[7] = _objectReader.ReadBitDouble();
-				matrix[8] = _objectReader.ReadBitDouble();
-				matrix[9] = _objectReader.ReadBitDouble();
-				matrix[10] = _objectReader.ReadBitDouble();
-				matrix[11] = _objectReader.ReadBitDouble();
-				matrix[12] = _objectReader.ReadBitDouble();
-				matrix[13] = _objectReader.ReadBitDouble();
-				matrix[14] = _objectReader.ReadBitDouble();
-				matrix[15] = _objectReader.ReadBitDouble();
+				double[] elements  = new double[16];
+				elements[0] = _objectReader.ReadBitDouble();
+				elements[1] = _objectReader.ReadBitDouble();
+				elements[2] = _objectReader.ReadBitDouble();
+				elements[3] = _objectReader.ReadBitDouble();
+				elements[4] = _objectReader.ReadBitDouble();
+				elements[5] = _objectReader.ReadBitDouble();
+				elements[6] = _objectReader.ReadBitDouble();
+				elements[7] = _objectReader.ReadBitDouble();
+				elements[8] = _objectReader.ReadBitDouble();
+				elements[9] = _objectReader.ReadBitDouble();
+				elements[10] = _objectReader.ReadBitDouble();
+				elements[11] = _objectReader.ReadBitDouble();
+				elements[12] = _objectReader.ReadBitDouble();
+				elements[13] = _objectReader.ReadBitDouble();
+				elements[14] = _objectReader.ReadBitDouble();
+				elements[15] = _objectReader.ReadBitDouble();
+
+				annotContext.TransformationMatrix = new Matrix4(elements);
```

Comment:
It would significantly reduce memory impact here (128 bytes for a double[16] per each object read) if you used the Matrix4 constructor below.
This is an an issue especially in this hot path because the double[] is just discarded anyway.

`public Matrix4(
			double m00, double m10, double m20, double m30,
			double m01, double m11, double m21, double m31,
			double m02, double m12, double m22, double m32,
			double m03, double m13, double m23, double m33)`

Reply:
OK